src/components/LogsTab/index.tsx

```diff
@@ -79,19 +79,18 @@ export class LogsTab extends React.PureComponent<Props, State> {
         this.setState({ isStopped });
       }
 
-      if (workspacesLogs) {
-        const logs = workspacesLogs.get(workspaceId);
-        if (logs && (logs.length !== this.state.logs.length)) {
-          this.setState({ logs });
-        }
+      const logs = workspacesLogs.get(workspaceId);
+      if (this.state.logs !== logs) {
```

Comment:
I don't think you can compare objects of type `string[] | undefined` in this way.

Reply:
From offline discussion: it's comparing of internal references of two objects. A comment will be added to avoid confusing.